eslint-plugin-expensify/prefer-at.js

```diff
@@ -0,0 +1,138 @@
+                    const right = parseExpression(node.right);
+                    if (left !== null && right !== null) {
+                        return `(${left} ${node.operator} ${right})`;
+                    }
+                    return null;
+
+                case 'UnaryExpression':
+                    const argument = parseExpression(node.argument);
+                    if (argument !== null) {
+                        return `${node.operator}${argument}`;
+                    }
+                    return null;
+
+                case 'MemberExpression':
+                    if (node.property.type === 'Identifier' && node.property.name === 'length') {
+                        return `${getSourceCode(node.object)}.length`;
+                    }
+                    return null;
+
+                case 'Identifier':
+                    return node.name;
+
+                default:
+                    return null;
+            }
+        }
+
+        function getExpressionWithUpdatedBrackets(indexExpression) {
+            if (indexExpression.startsWith('(') && indexExpression.endsWith(')')) {
+                return indexExpression.slice(1, -1);
+            }
+            return indexExpression;
+        }
+
+        function checkNode(node) {
+            if (node.type === 'MemberExpression' && node.property) {
+                if (!isArrayType(node.object)) {
+                    return;
+                }
+
+                const indexExpression = parseExpression(node.property);
+
+                if (indexExpression !== null && indexExpression !== 'length' && indexExpression !== 'at') {
+                    // Check for specific pattern a[a.length - y]
+                    if (
+                        node.property.type === 'BinaryExpression' &&
+                        node.property.operator === '-' &&
+                        isMemberExpressionWithLength(node.property.left) &&
+                        isLiteralNumber(node.property.right)
+                    ) {
+                        const offset = node.property.right.value;
+                        context.report({
+                            node,
+                            message,
+                            fix(fixer) {
+                                const objectText = getSourceCode(node.object);
+                                return fixer.replaceText(node, `${objectText}.at(-${offset})`);
+                            },
+                        });
+                    } else {
+                        context.report({
+                            node,
+                            message,
+                            fix(fixer) {
+                                const objectText = getSourceCode(node.object);
+                                return fixer.replaceText(node, `${objectText}.at(${getExpressionWithUpdatedBrackets(indexExpression)})`);
+                            },
+                        });
+                    }
```

Comment:
```suggestion
                    context.report({
                        node,
                        message,
                        fix(fixer) {
                            const objectText = getSourceCode(node.object);
                            return fixer.replaceText(node, `${objectText}.at(${getExpressionWithUpdatedBrackets(indexExpression)})`);
                        },
                    });
```

Do you think we really need this `a[a.length - 1]` to `a.at(-1)` use case? Seems overly specific to me and not too beneficial, it's a optimisation that the developer can apply by itself. Also, it's currently converting this wrong use case:

```ts
        {
            code: `
                const sample = [1, 2, 3, 4];
                sample[a.length - 1] // Notice that i'm using a different variable "randomVar".
                `,
            output: `
                const sample = [1, 2, 3, 4];
                sample.at(-1)
                `,
            errors: [{
                message,
            }],
        },
```